packages/hash/frontend/src/blocks/page/BlockHighlight.tsx

```diff
@@ -0,0 +1,41 @@
+import { useTheme } from "@mui/material";
+import { motion } from "framer-motion";
+import { useBlockHighlightContext } from "./BlockHighlightContext";
+import { useBlockView } from "./BlockViewContext";
+
+interface HighlightProps {
+  onAnimationComplete: () => void;
+}
+
+const Highlight = ({ onAnimationComplete }: HighlightProps) => {
+  const theme = useTheme();
+
+  return (
+    <motion.div
+      animate={{ opacity: 0 }}
+      transition={{ duration: 4 }}
+      onAnimationComplete={onAnimationComplete}
+      style={{
+        position: "absolute",
+        width: 6,
+        top: 6,
+        bottom: 6,
+        borderRadius: 3,
+        left: "1rem",
+        background: theme.palette.blue[70],
+      }}
+    />
+  );
+};
+
+export const BlockHighlight = () => {
+  const blockView = useBlockView();
+  const { highlightedBlockId, setHighlightedBlockId } =
+    useBlockHighlightContext();
+
+  if (!highlightedBlockId || highlightedBlockId !== blockView.dom.id) {
```

Comment:
Since this component is rendered directly by `BlockView`, you could pass `blockEntityId` as a prop instead of relying on the block view context. 

To me, looking at this component it's a bit easier thinking about what a `blockEntityId` prop is versus `blockView.dom.id` which isn't as descriptive / obvious.

It also more clearly decouples the component from a specific context, e.g. we can move it somewhere else, or update the properties of `BlockView` or the shape of `useBlockView` without affecting this – it just needs a block entity id to function.

This is personal preference though so feel free to disagree.

Reply:
I agree with @CiaranMn . It will be easier to understand if we pass the `blockEntityId` as a prop, then we can use `getBlockDomId` util method to get the actual dom id. 

An example of it's usage is in BlockLoadedProvider

https://github.com/hashintel/hash/blob/5cadd6ec35428cd58afd55a4bd007361df4fed43/packages/hash/frontend/src/blocks/onBlockLoaded.tsx#L48

Reply:
Used `blockEntityId` instead of `blockView.dom.id` in https://github.com/hashintel/hash/pull/914/commits/f96ddefe716f73b7d95483e246e19a86c32c9540